pmd-core/src/main/java/net/sourceforge/pmd/properties/AbstractPropertySource.java

```diff
@@ -48,6 +48,15 @@ public void definePropertyDescriptor(PropertyDescriptor<?> propertyDescriptor) {
         propertyDescriptors.add(propertyDescriptor);
     }
 
+    public final void definePropertyDescriptorFinal(PropertyDescriptor<?> propertyDescriptor) {
+        // Check to ensure the property does not already exist.
+        if (getPropertyDescriptor(propertyDescriptor.name()) != null) {
+            throw new IllegalArgumentException("There is already a PropertyDescriptor with name '"
+                    + propertyDescriptor.name() + "' defined on " + getPropertySourceType() + " " + getName() + ".");
+
+        }
+        propertyDescriptors.add(propertyDescriptor);
+    }
```

Comment:
what's the point of doing this? This is a verbatim copy of the existing `definePropertyDescriptor`. Other than adding duplicate code we need to maintain this achieves nothing.